### DISCUSSION TITLE
Cordova & Background functionality

### DISCUSSION THREAD

**Discussion by EinfachHans** [socketio/socket.io#3798] [Q&A] [answered]
Hey there,

we are currently using WebSockets in on of our applications, but the used libraries seems to have some problems so we are thinking about switching to `socket.io`.

Or app is written in Cordova and in cooperation with the german police. As an emergency app the functionality in the background is very very important. [Here](https://socket.io/socket-io-with-apache-cordova) it says for Cordova you can use the JS Client, but what about the background functionality if the app is closed (in task manager open) for like an hour? We use several plugins to keep the app itself open which works quite well.

Any experience, any infos about that?

**Comment by darrachequesne** [comment, marked as answer]
Keeping a WebSocket connection open in the background will likely result in a quick battery drain, so I don't think Socket.IO is the best option for this use case.

Did you check [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/)?

**Reply by EinfachHans** [reply]
@darrachequesne Yeah that's correct. We currently also have a hige battery usage wich is okay for us and this app.

Yeah that is the thing, we are currently thinking about either try to fix our websocket problem or rebuild everything with push messages from server -> app and basic http requests from app -> server.

So the second option is better in your oppinion?

**Reply by darrachequesne** [reply]
Yes, I think FCM is the de-facto solution for background notifications.
